junifer/storage/hdf5.py

```diff
@@ -0,0 +1,954 @@
+            raise_error(
+                msg=(
+                    "At least one of `feature_name` or `feature_md5` "
+                    "must be specified."
+                )
+            )
+        # Parameter check pass
+        else:
+            # Read metadata
+            metadata = read_hdf5(
+                fname=str(self.uri.resolve()),  # type: ignore
+                title="meta",
+                slash="ignore",
+            )
+            # Initialize MD5 variable
+            md5: str = ""
+
+            # Consider feature_md5
+            if feature_md5 and not feature_name:
+                logger.debug(
+                    f"Validating feature MD5 '{feature_md5}' in metadata "
+                    f"for: {self.uri.resolve()} ..."  # type: ignore
+                )
+                # Validate MD5
+                if feature_md5 in metadata:
+                    md5 = feature_md5  # type: ignore
+                else:
+                    raise_error(msg=f"Feature MD5 '{feature_md5}' not found")
+
+            # Consider feature_name
+            elif not feature_md5 and feature_name:
+                logger.debug(
+                    f"Validating feature name '{feature_name}' in metadata "
+                    f"for: {self.uri.resolve()} ..."  # type: ignore
+                )
+                # Retrieve MD5 for feature_name
+                # Implicit counter for duplicate feature_name with different
+                # MD5; happens when something is wrong with marker computation
+                feature_name_duplicates_with_different_md5 = []
+                for md5, meta in metadata.items():
+                    if meta["name"] == feature_name:
+                        feature_name_duplicates_with_different_md5.append(md5)
+
+                # Check for no / duplicate feature_name
+                if len(feature_name_duplicates_with_different_md5) == 0:
+                    raise_error(msg=f"Feature '{feature_name}' not found")
+                elif len(feature_name_duplicates_with_different_md5) > 1:
+                    raise_error(
+                        msg=(
+                            "More than one feature with name "
+                            f"'{feature_name}' found. You can bypass this "
+                            "issue by specifying a `feature_md5`."
+                        )
+                    )
+
+                md5 = feature_name_duplicates_with_different_md5[0]
+
+            # Read data from HDF5
+            hdf_data = read_hdf5(
+                fname=str(self.uri.resolve()),  # type: ignore
+                title=md5,
+                slash="ignore",
+            )
+            reshaped_data = hdf_data["data"]
+
+            # Generate index for the data
+            logger.debug(f"Generating pandas.MultiIndex for {md5} ...")
+            # Create dictionary for aggregating index data
+            element_idx_dict = defaultdict(list)
+
+            if hdf_data["kind"] == "matrix":
```

Comment:
The matrix in the DF format should be flattened. Indeed check what we do with matrices before storing in SQLite. This is what needs to happen now, here.

Reply:
Maybe I'm not getting the rationale here, what do you mean?

Reply:
When you convert a matrix to dataframe, it must generate one row per element. As a general rule, each DF must have only one row per element (except for timeseries).

Reply:
Should be addressed in the latest commits.